bundles/tools.vitruv.applications.demo.familiespersons/src/tools/vitruv/applications/demo/familiespersons/families2persons/FamiliesToPersons.reactions

```diff
@@ -36,8 +37,9 @@ routine deletePersonRegister(families::FamilyRegister familyRegister) {
 	match {
 		val personRegister = retrieve persons::PersonRegister corresponding to familyRegister
 	}
-	action {
-		delete personRegister
+	update {
+		deleteObject(personRegister)
+		removeCorrespondenceBetween(personRegister, familyRegister)
```

Comment:
Maybe this was an upfront design decision to not do so (and I think it was already discussed somewhen) but wouldn't it be easier that deleting an object also deletes all of its correspondences?
At least all case studies I am aware of expect this behaviour.

Reply:
Currently there is a validation in the `ChangePropagator` that all correspondences of a deleted element must have been removed. Otherwise an exception is thrown. This is supposed to ensure that developers are aware of all corresponding elements and handle them appropriately. E.g., let `a1` and `a2` correspond to `b`, and `a1` is removed such that `b` and the correspondence to it are also removed, then the correspondence to `a2` should not be removed, but it should be indicated that something has to be done with `a2` as well. This is why the current behavior is that correspondences to all elements that have been retrieved in a routine are removed implicitly when deleting an element. However, I found this behavior not very intuitive (not to say confusing when changing the language), such that I made correspondence removal explicit, just like their addition. But this should, of course, be discussed.

Reply:
But is this "sanity" check something that we should consider when designing the language? For me this sounds rather like a feature we could turn on when profiling / debugging some application. At least right now it is unintuitive for me that `deleteObject` deletes the object with all its references in the V-SUM models, but not its references in the correspondence model.

Reply:
I see your point, but then we should discuss whether it should be a general "feature" of the change propagation and not of the Reactions language, as it should not depend on the language used to define a `ChangePropagationSpecification` whether correspondences are removed automatically or not.

Still, I think it should not be default behavior to have _all_ correspondences removed. We did not have that behavior in the language before (but just removing the correspondences we have retrieved in the routine) because of scenarios like the one I sketched above. If you simply remove all correspondences, then there is no chance that the transformation in the other direction can react to the changes properly, as all correspondences have been removed, even those that were only relevant for the opposite transformation direction.

Reply:
Since `deleteObject` currently actually performs a remove operation (i.e., calls `EcoreUtil.remove`) because of incompatibilities of `EcoreUtil.delete` with the UML2 metamodel, we have decided to rename the method to `removeObject`. Then there should not be the expectation that it also deletes correspondences. We have agreed that completely manual correspondence removal is the best option right now, as corresponces to not have a well-defined semantics but can be used by Reactions ad-hoc, such that there is not generic mechanism to decide which correspondences to remove and when. In case this changes in the future, we may have a mechanism that removes correspondences automatically under well-defined conditions.